mdp/nodes/sfa_nodes.py

```diff
@@ -68,10 +72,37 @@ class SFANode(Node):
 
           You can even change this behaviour during training. Just set the
           corresponding switch in the `train` method.
+
+
+      ``handle_rank_deficit``
+          If ``True`` the `stop_train` method solves the SFA eigenvalue
+          problem in a way that is robust against linear redundancies in
+          the input data. This leads to rank deficit in the covariance
+          matrix, which usually yields a
+          SymeigException ('Covariance matrices may be singular').
+          There are several solving methods implemented, which you can
+          select by setting the _sfa_solver field, e.g.
+
+             sfa = SFANode()
+             sfa._sfa_solver = sfa._rank_deficit_solver_pca
+
+          Currently there are three solvers available:
+
+          _rank_deficit_solver_reg    - works by regularization
+          _rank_deficit_solver_pca - works by PCA
+          _rank_deficit_solver_svd    - works by svd
```

Comment:
We should add at least some information about how one would choose a method in case `'symeig'` fails, i.e., why choose regularization over PCA etc. We might have to wait until we have more benchmark results for this though.

Reply:
I'm perfectly fine with extending this doc.

Reply:
As I mentioned somewhere else (the number of mini-threads is becoming unwieldy) I'd suggest once the code is feature complete, we make a documentation and usability iteration of your changes in person and then post it here for (hopefully final) review.